helpers/etcd.py

```diff
@@ -45,7 +46,7 @@ def put_client_path(self, path, data):
             base64string = base64.encodestring('%s:%s' % (self.authentication["username"], self.authentication["password"])).replace('\n', '')
             request.add_header("Authorization", "Basic %s" % base64string)
         request.get_method = lambda: 'PUT'
-        urllib2.urlopen(request).read()
+        urllib2.urlopen(request, timeout=self.timeout).read()
```

Comment:
`urlopen` uses the global timeout settings by default. But this way we can use the value from the config to limit it so we aren't hanging on trying to contact what should be a local etcd for forever